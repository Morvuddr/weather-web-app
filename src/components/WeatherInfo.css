.WeatherInfo{
    width: 45%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}
.Info{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    height: 40px;
    background: gray;
}
.Info h3{
    margin: 0px;
    padding-left: 10px;
    color: white;
}
.Info h4{
    margin: 0px;
    padding-right: 10px;
    color: lightgray;
}