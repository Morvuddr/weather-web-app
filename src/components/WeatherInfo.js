import React from 'react';
import '../styles/App.css';
import * as PropTypes from 'prop-types';

const WeatherInfo = ({ weather }) => (
    !!weather ? (
            <div className="WeatherInfo">
                <div className="Info">
                    <h3>Ветер</h3>
                    <h4>{weather.wind}</h4>
                </div>
                <div className="Info">
                    <h3>Облачность</h3>
                    <h4>{weather.cloudiness}</h4>
                </div>
                <div className="Info">
                    <h3>Давление</h3>
                    <h4>{weather.pressure}</h4>
                </div>
                <div className="Info">
                    <h3>Влажность</h3>
                    <h4>{weather.humidity}</h4>
                </div>
                <div className="Info">
                    <h3>Координаты</h3>
                    <h4>{weather.location}</h4>
                </div>
            </div>
        ) :
        null
);

WeatherInfo.propTypes = {
    weather: PropTypes.object
};

export default WeatherInfo;