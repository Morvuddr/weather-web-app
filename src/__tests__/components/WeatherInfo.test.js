import WeatherInfo from '../../components/WeatherInfo';
import renderer from 'react-test-renderer';
import React from 'react';

describe('WeatherInfo component', () => {

    test('matches the snapshot when weather != null', () => {
        const city = {
            id: '1',
            name: 'Moscow',
            img: 'abcd',
            temperature: 'temp',
            wind: 'wind',
            cloudiness: 'cloudiness',
            pressure: 'pressure',
            humidity: 'humidity',
            location: '[location]'
        };
        const tree = renderer.create(
           <WeatherInfo weather={city}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('matches the snapshot when weather == null', () => {

        const tree = renderer.create(
            <WeatherInfo weather={null}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});