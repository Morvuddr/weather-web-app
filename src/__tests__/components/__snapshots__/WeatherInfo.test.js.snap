// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherInfo component matches the snapshot when weather != null 1`] = `
<div
  className="WeatherInfo"
>
  <div
    className="Info"
  >
    <h3>
      Ветер
    </h3>
    <h4>
      wind
    </h4>
  </div>
  <div
    className="Info"
  >
    <h3>
      Облачность
    </h3>
    <h4>
      cloudiness
    </h4>
  </div>
  <div
    className="Info"
  >
    <h3>
      Давление
    </h3>
    <h4>
      pressure
    </h4>
  </div>
  <div
    className="Info"
  >
    <h3>
      Влажность
    </h3>
    <h4>
      humidity
    </h4>
  </div>
  <div
    className="Info"
  >
    <h3>
      Координаты
    </h3>
    <h4>
      [location]
    </h4>
  </div>
</div>
`;

exports[`WeatherInfo component matches the snapshot when weather == null 1`] = `null`;
